src/status_im/ui/screens/accounts/login/events.cljs

```diff
@@ -40,27 +43,73 @@
                        :name name)
      :dispatch [:navigate-to :login]}))
 
+(defn wrap-with-login-account-fx [db address password]
+  {:db     db
+   ::login [address password]})
+
+(register-handler-fx
+  ::login-account
+  (fn [{db :db} [_ address password]]
+    (wrap-with-login-account-fx
+      (assoc db :node/after-start nil)
+      address password)))
+
+(defn get-network-by-address [db address]
+  (let [accounts (get db :accounts/accounts)
+        {:keys [network networks]} (get accounts address)
+        config   (get-in networks [network :config])]
+    {:network network
+     :config  config}))
+
+(defn wrap-with-initialize-geth-fx [db address password]
+  (let [{:keys [network config]} (get-network-by-address db address)]
+    {:initialize-geth-fx config
+     :db                 (assoc db :network network
+                                   :node/after-start [::login-account address password])}))
+
+(register-handler-fx
+  ::start-node
+  (fn [{db :db} [_ address password]]
+    (wrap-with-initialize-geth-fx
+      (assoc db :node/after-stop nil)
+      address password)))
+
+(defn wrap-with-stop-node-fx [db address password]
+  {:db         (assoc db :node/after-stop [::start-node address password])
+   ::stop-node nil})
+
 (register-handler-fx
   :login-account
-  (fn [{db :db} [_ address password account-creation?]]
-    {:db     (-> db
-                 (assoc :accounts/account-creation? account-creation?)
-                 (assoc-in [:accounts/login :processing] true))
-     ::login [address password]}))
+  (fn [{{:keys [network status-node-started?] :as db} :db}
+       [_ address password account-creation?]]
+    (let [{account-network :network} (get-network-by-address db address)
+          db' (-> db
```

Comment:
Seems to do a lot, put in separate fn and another name than db'?

Reply:
+1